specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/virtualNetwork.json

```diff
@@ -1454,6 +1454,13 @@
           },
           "description": "Array of IP configuration profiles which reference this subnet."
         },
+        "ipAllocations": {
+          "type": "array",
+          "items": {
+            "$ref": "./ipAllocation.json#/definitions/IpAllocation"
```

Comment:
>./ipAllocation.json#/definitions/IpAllocation" [](start = 21, length = 46)

this should only reference the resource ID, not contain the full resource definition again.  Otherwise, /read RBAC is broken

Reply:
Could you please give suggestion about how to specify the reference as ID here? I search swagger files and document didn't how to reference the resource ID here. 

When we run the generated SDK, the result here is resource ID, not the full resource definition, so the RBAC is not broken here.

Reply:
See how this is referencing "natGateway" above.  network.json has a SubResource definition for referencing by ID

Reply:
Thanks, fixed with SubResource